CIP-0026/CIP-0026.md

```diff
@@ -0,0 +1,468 @@
+
+## Decentralization
+
+For much of the metadata we are concerned with there is no “right” answer. The metadata server is thus playing a key trusted role - even if that trust is partial because users can rely on attestation signatures. We therefore believe that it is critical that we allow users to choose which metadata server (or servers) they refer to.
+
+An analogy is with DNS nameservers: these are a trusted piece of infrastructure for any particular user, but users have a choice of which nameservers to use, and can use multiple.
+
+This also makes it possible for these servers to be a true piece of community infrastructure, rather than something wedded to a major player (although we hope that the Cardano Foundation and IOHK will produce a competitive offering).
+
+## Verifiable vs non-verifiable metadata
+
+A key distinction is between metadata that is verifiably correct and that which is non-verifiable.
+
+The key example of verifiable metadata is hashe pre-images. Where the metadata subject is a hash, the preimage of that hash is verifiable metadata, since we can simply hash it and check that it matches. This covers several cases that we are interested in:
+
+- Mapping script hashes to the script
+- Mapping public key hashes to the public key
+
+However, most metadata is non-verifiable:
+
+- Human-readable names for scripts are not verifiable: there is no “right” answer to what the name of a script is.
+- Many scripts do not have an obvious “owner” (certainly this is true for Plutus scripts), but even for those which do (e.g. phase-1 monetary scripts) this does not mean that the owner is trusted! See the “Security” section below for more discussion on trust.
+- Any associations with authors, websites, icons, etc are similarly non-verifiable as there is no basis for establishing trust.
+
+## Security
+
+Most of the security considerations relate to non-verifiable metadata. Verifiable metadata can generally always be accepted as is (provided that it is verified). Our threat model is that non-verifiable metadata may always have been provided by an attacker.
+
+Accepting non-verifiable metadata blindly can lead to attacks. For example, a malicious server or user might attempt to name their currency “Ada”. If we blindly accept this and overwrite the existing mapping for “Ada”, this would lead to easy phishing attacks. The approach we take is heavily inspired by petname systems, GPG keyservers, and local address books. The user always has a local mapping which is trusted, and then adding to or updating that mapping requires explicit user consent, unless we can prove that this is trustworthy (i.e. the metadata is verifiable).
+
+How can users decide whether to trust an update? This is where the attestation signatures come into play. If a user trusts the entity which signs the metadata record, that may be sufficient for them to accept it as a legitimate update. See the earlier “Trust” sections for some discussion of trusted keys and updates.
```

Comment:
There is no "Trust" section.

Reply:
Ah yes indeed, it was a bit re-structured. I'll adjust.